sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2213,6 +2213,14 @@ object SQLConf {
       .intConf
       .createWithDefault(1)
 
+  val STREAMING_STATE_STORE_USE_AVRO_ENCODING =
+    buildConf("spark.sql.streaming.stateStore.useAvroEncoding")
```

Comment:
maybe rename as 

`stateStore.encodingFormat`

and use default as `UnsafeRow` and also allow for setting it to `Avro`